jupyter_sphinx/execute.py

```diff
@@ -308,11 +351,24 @@ def apply(self):
             kernel_name = nodes[0]['kernel_name'] or default_kernel
             notebook_name = nodes[0]['notebook_name'] or next(default_names)
 
-            notebook = execute_cells(
+            notebook, widget_state = execute_cells(
                 kernel_name,
                 [nbformat.v4.new_code_cell(node.astext()) for node in nodes],
                 self.config.jupyter_execute_kwargs,
             )
+
+            if widget_state:
+                # Append widget state JSON if any widgets were used in the notebook.
+                # XXX: Can we specify a javascript node directly, rather than a 'raw'
+                #      node of 'html' format?
+                doctree.append(docutils.nodes.raw(
```

Comment:
This can potentially be a large amount of data; should we perhaps save it as a separate file and provide it via `src=...`?

Reply:
We may do. `embed_widgets` does not do this currently.

Reply:
I think it was designed before heavy-weight widgets like `PlotlyWidget`. Hm, but even the core library has e.g. `ImageWidget`, whose state shouldn't just be embedded into the page.

Reply:
Fixed in 766dfb79ea707a6f37b888d95f316796f77330c5